java/vector/src/main/java/org/apache/arrow/vector/util/DateUtility.java

```diff
@@ -664,29 +662,46 @@ public static DateTimeFormatter getDateTimeFormatter() {
   // Function returns time formatter used to parse time strings
   public static DateTimeFormatter getTimeFormatter() {
     if (timeFormat == null) {
-      DateTimeFormatter timeFormatter = DateTimeFormat.forPattern("HH:mm:ss");
-      DateTimeParser optionalSec = DateTimeFormat.forPattern(".SSS").getParser();
+      DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("HH:mm:ss");
+      DateTimeFormatter optionalSec = DateTimeFormatter.ofPattern(".SSS");
       timeFormat = new DateTimeFormatterBuilder().append(timeFormatter).appendOptional(optionalSec).toFormatter();
     }
     return timeFormat;
   }
 
-  public static int monthsFromPeriod(Period period) {
-    return (period.getYears() * yearsToMonths) + period.getMonths();
-  }
-
-  public static int millisFromPeriod(final Period period) {
-    return (period.getHours() * hoursToMillis) +
-        (period.getMinutes() * minutesToMillis) +
-        (period.getSeconds() * secondsToMillis) +
-        (period.getMillis());
+  /**
+   * Convert milliseconds from epoch to a LocalDateTime with UTC offset
+   *
+   * @param epochMillis milliseconds from epoch
+   * @return LocalDateTime object with UTC offset
+   */
+  public static LocalDateTime getLocalDateTimeFromEpochMilli(long epochMillis) {
```

Comment:
rather than duplicate these lines in vector and holder reader classes, I just put them here